platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/VirtualLeafRecordSerializer.java

```diff
@@ -97,14 +107,17 @@ public VirtualLeafRecord<K, V> deserialize(final ByteBuffer buffer, final long d
         final DataItemHeader dataItemHeader = deserializeHeader(buffer);
         // deserialize path from the header
         final long path = dataItemHeader.getKey();
-        // deserialize hash
-        final Hash hash = byteBufferToHash(buffer, hashSerializationVersion);
+
+        if (hashSerializationVersion != 0) {
+            // compatibility: read hash
+            byteBufferToHash(buffer, hashSerializationVersion);
```

Comment:
Does `byteBufferToHash()` actually create a Hash object? This is redundant, it would be enough to just read 48 (default digest type length) bytes from the buffer and throw them away.

Reply:
> It would be great, if serializers could read data according to its version
This is exactly how it works. However, there is one weird corner case:
- `VirtualLeafRecordSerializer` is created by deserialization. In this case it's still has hash-related bits in its version
- it serializes values on rehashing

If I exclude the empty hash, the file format will be corrupted because the hash is expected there. It shouldn't be a problem because the hash is not used anyway. Next time this serializer is deserialized it will not have hash-related bytes in its version and it won't have to persist empty hashes. 

There is only one danger to that - this change is not backward compatible. Once the node starts, the hashes will be corrupted in the old storage which shouldn't be a big deal since the hashes are data derivatives and can be recalculated.

Reply:
I replaced it with just skipping bytes.